backend/app/api/routers/threads.py

```diff
@@ -0,0 +1,68 @@
+from datetime import datetime
+from http import HTTPStatus
+from typing import Optional
+
+from fastapi import APIRouter, WebSocket, WebSocketDisconnect
+
+from app.api.exceptions import EntityNotFound
+from app.api.managers import WsConnectionManager
+from app.api.schemas import Filter, MessageCreate, Sender, SupportCreate, SupportThread
+from app.api.services import SupportThreadService
+
+router = APIRouter()
+ws_manager = WsConnectionManager()
+
+
+@router.post('/threads', response_model=SupportThread, status_code=HTTPStatus.CREATED)
+async def create_thread(thread: SupportCreate) -> SupportThread:
+    thread_new = await SupportThreadService.create(thread)
+
+    return thread_new
+
+
+@router.get('/threads', response_model=list[SupportThread])
+async def find_all_threads(flt: Optional[Filter] = None) -> list[SupportThread]:
+    threads = await SupportThreadService.find_all(flt)
+
+    return threads
+
+
+@router.get('/threads/{thread_id}', response_model=SupportThread)
+async def find_thread(thread_id: int) -> SupportThread:
+    thread = await SupportThreadService.find_by_id(thread_id)
+
+    if not thread:
+        raise EntityNotFound('thread')
+
+    return thread
+
+
+@router.put('/threads/{thread_id}', response_model=SupportThread)
+async def update_thread(thread_id: int, thread: SupportCreate) -> SupportThread:
+    thread_upd = await SupportThreadService.update(thread, thread_id)
+
+    return thread_upd
```

Comment:
The purpose of the method is to update support thread, since we will have redirect method(redirect question to another volunteer), we should replace volunteer_id field. I will replace it to patch without allowing redefine client_id, id and messages

Reply:
Nevertheless, it is currently possible to rewrite the messages history with this method, isn't it? This does not seem right to me.